is.ts

```diff
@@ -751,6 +751,129 @@ type IsTupleOfMetadata = {
   args: [Parameters<typeof isTupleOf>[0], Parameters<typeof isTupleOf>[1]?];
 };
 
+/**
+ * Return a type predicate function that returns `true` if the type of `x` is `ParametersOf<T>` or `ParametersOf<T, E>`.
+ *
+ * This is similar to `TupleOf<T>` or `TupleOf<T, E>`, except that the `OptionalOf<P>` at the end of the tuple becomes optional.
```

Comment:
I think it is not clear what happen when `OptionalOf` exists in the middle of the tuple with this explanation. It is not must but should be explained a bit more clearly.

For example, like this

```ts
const isMyType = is.ParametersOf([
  is.Number,
  is.OptionalOf(is.String),
  is.Boolean,
  is.OptionalOf(is.Number),
  is.OptionalOf(is.String),
  is.OptionalOf(is.Boolean),
]);

// maybe this?
// type isMyType = Predicate<[number, string | undefined, boolean, number?, string?, boolean?]>;
```

Reply:
Fix the doc and examples. How about this?

Reply:
Ah... mistake commit.